partner_communication_switzerland/wizards/portal_wizard.py

```diff
@@ -9,25 +9,60 @@
 #
 ##############################################################################
 
+import logging
+
 from odoo import api, models
+from odoo.tools.translate import _
+from odoo.exceptions import UserError
+
+
+_logger = logging.getLogger(__name__)
 
 
 class PortalWizard(models.TransientModel):
     _inherit = 'portal.wizard'
 
+
+class PortalWizardUser(models.TransientModel):
+    _inherit = 'portal.wizard.user'
+
     @api.multi
     def action_apply(self):
-        """
-        Generate communication for User invitation
-        """
-        res = super(PortalWizard, self).action_apply()
-        comm_obj = self.env['partner.communication.job']
-        config = self.env.ref(
-            'partner_communication_switzerland.portal_welcome_config')
-        for user in self.mapped('user_ids.user_id'):
-            comm_obj.create({
-                'partner_id': user.partner_id.id,
-                'object_ids': user.id,
-                'config_id': config.id
-            })
+        res = super(PortalWizardUser, self).action_apply()
+
+        # to not create a communication
+        # you should give context
+        # { 'create_uid_communication': False }
+        # else it will create a communication
+        if self.env.context['create_communication_with_uid'] is None:
+            bool_communication_with_uid = True
+        else:
+            bool_communication_with_uid = \
+                self.env.context['create_communication_with_uid']
+
+        if bool_communication_with_uid:
+            self._create_uid_communication()
```

Comment:
Here you can add inside res the communication created (or just the id maybe if it doesn't work). `res['communication'] = self._create_uid_communication()`. `_create_uid_communication` has of course to return the communication record.